### DISCUSSION TITLE
Setting up AI-flow on MacOS with Docker

### DISCUSSION THREAD

**Discussion by wecameinpeace** [DahnM20/ai-flow#8] [Q&A]
Hi Dahn,

I am currently setting up the AI-Flow application to work with both the Replicate API and the OpenAI ChatGPT API. I’ve integrated the necessary API keys into the Docker environment but it's still not working:

1. API Integration Confirmation:
   - Could you please confirm whether the environment variable names I used for the APIs (`OPENAI_API_KEY`, `REPLICATE_API_KEY`) are correct and fully configured within the backend code? Is there any other API Variable missing, like for AI-Flow itself?
   - Are there any additional configurations or steps required to ensure that these APIs are properly connected and authenticated within our application’s backend?

2. Serving & Connectivity to Web:
   - What additional steps should be taken to connect our application to the web for the API connection? Are there specific configurations for production deployment that I should be aware of to ensure security and efficiency?
   - How is the server configured to handle requests to and from these APIs, and is there anything I should adjust or optimize?

3. Credits Management with AI-Flow:
   - We bought some credits at AI-Flow, can they be used with the local App – Could you guide how this is currently configured? or are they just ment for your online version? https://app.ai-flow.net/

Best regards,
Martin, DV5 Vienna

**Comment by DahnM20** [comment]
Hi Martin,

Sorry for the late answer, I just saw your post now.

1. By default, the keys used to run the nodes are the ones specified by the user in the UI of the app. If you want to use your env keys for every request, you must add `USE_ENV_API_KEYS=true` in the .env file of the backend. 

     _(You may need to specified a dummy value in the OpenAI API Key field in the UI though. Otherwise the UI will show an error message and not send requests to the server, I will improve this in future versions.)_

3. Locally, you have to be careful when uploading files. Some files are stored directly locally by default (for example, the Stable Diffusion or the OpenAI TTS node). In this case, you will not be able to send them to other APIs. For that, you will have to change the configuration slightly. There's a new page in the documentation about that: [File Upload](https://docs.ai-flow.net/docs/file-upload/file-upload-s3).

     _N.B.: This does not apply to the API which directly returns files as URLs, such as Replicate or DALL-E._

4. Credits are only for the online version; the local app is a simplified version of the online version, designed to be used as a personal tool.

Please keep me updated,

Best regards,
Dahn

**Comment by wecameinpeace** [comment]
Hi Dahn,

Thank you very much for your reply!

Great, I’ll try that locally and eventually stick more to the online version.

Just one litte detail: in the online version, using Chrome on MacOS the text results inside the GPT nodes are dark grey on black,
I think I have seen them somewhere in your docs in white on black. (enclosed a screenshot)

And – as a designer and 3D guy I’m very happy for your Node Interface, that’s really a great approach!
Thank you very much for that in particular.
as a fan – If you would need anything I'd be happy to help you on the design or brand side.

Best regards,
Martin



> Am 06.05.2024 um 08:14 schrieb Dahn ***@***.***>:
> 
> 
> Hi Martin,
> 
> Sorry for the late answer, I just saw your post now.
> 
> By default, the keys used to run the nodes are the ones specified by the user in the UI of the app. If you want to use your env keys for every request, you must add USE_ENV_API_KEYS=true in the .env file of the backend.
> 
> Locally, you have to be careful when uploading files. Some files are stored directly locally by default (for example, the Stable Diffusion or the OpenAI TTS node). In this case, you will not be able to send them to other APIs. For that, you will have to change the configuration slightly. There's a new page in the documentation about that: File Upload <https://docs.ai-flow.net/docs/file-upload/file-upload-s3>.
> 
> N.B.: This does not apply to the API which directly returns files as URLs, such as Replicate or DALL-E.
> 
> Credits are only for the online version; the local app is a simplified version of the online version, designed to be used as a personal tool.
> 
> Please keep me updated,
> 
> Best regards,
> Dahn
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/DahnM20/ai-flow/discussions/8#discussioncomment-9324894>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/BEQTAQ7KWE3BQJYZDD5QT53ZA4NVLAVCNFSM6AAAAABHGR4H72VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TGMRUHA4TI>.
> You are receiving this because you authored the thread.
> 




￼

**Comment by DahnM20** [comment]
Feel free to ask about your local setup.

Yes, someone told me about the gray text yesterday too... I still can't reproduce it myself. But since you're using macOS, I'll try with that next!

Glad you like it! I liked the node system in Blender, even though I'm not very experienced with 3D, haha.

Contributions are welcome to make the app better for everyone. If you have any suggestions or ideas, feel free to reach out to me! You can find me on Twitter (@DahnM20), by email at dahn.m20.app@gmail.com, or on GitHub.
